lib/internal/errors.js

```diff
@@ -370,42 +370,38 @@ function makeSystemErrorWithCode(key) {
 }
 
 function makeNodeErrorWithCode(Base, key) {
-  return function NodeError(...args) {
-    const limit = Error.stackTraceLimit;
-    if (isErrorStackTraceLimitWritable()) Error.stackTraceLimit = 0;
-    const error = new Base();
-    // Reset the limit and setting the name property.
-    if (isErrorStackTraceLimitWritable()) Error.stackTraceLimit = limit;
-    const message = getMessage(key, args, error);
-    ObjectDefineProperties(error, {
-      [kIsNodeError]: {
-        __proto__: null,
-        value: true,
-        enumerable: false,
-        writable: false,
-        configurable: true,
-      },
-      message: {
-        __proto__: null,
-        value: message,
-        enumerable: false,
-        writable: true,
-        configurable: true,
-      },
-      toString: {
-        __proto__: null,
-        value() {
-          return `${this.name} [${key}]: ${this.message}`;
-        },
-        enumerable: false,
-        writable: true,
-        configurable: true,
-      },
-    });
-    captureLargerStackTrace(error);
-    error.code = key;
-    return error;
-  };
+  const defaultMessage = messages.get(key);
+  const expectedLength = (typeof defaultMessage !== 'string' && -1) || getExpectedArgumentLength(defaultMessage);
```

Comment:
Can this be simplified?
```suggestion
  const expectedLength = typeof defaultMessage !== 'string' ? -1 : getExpectedArgumentLength(defaultMessage);
```